src/TLuaInterpreter.cpp

```diff
@@ -309,6 +309,11 @@ int TLuaInterpreter::raiseEvent( lua_State * L )
             pE.mArgumentList.append( QString(lua_tostring( L, i )) );
             pE.mArgumentTypeList.append( ARGUMENT_TYPE_STRING );
         }
+        else if( lua_isboolean( L, i ) )
```

Comment:
Ah, it may just be numbers and strings that can get mixed up - a string "1234" can be understood as a number 1234 (it undergoes some sort of auto-conversion, or rather [coercion](http://www.lua.org/manual/5.1/manual.html#2.2.1)) without user intervention IIRC.